graphbolt/src/cuda/common.h

```diff
@@ -9,9 +9,67 @@
 #include <c10/cuda/CUDAException.h>
 #include <cuda_runtime.h>
 
+#include <memory>
+#include <unordered_map>
+
 namespace graphbolt {
 namespace cuda {
 
+/**
+ * @brief This class is designed to allocate workspace storage
+ * and to get a nonblocking thrust execution policy
+ * that uses torch's CUDA memory pool and the current cuda stream:
+ *
+ * cuda::CUDAWorkspaceAllocator allocator;
+ * const auto stream = torch::cuda::getDefaultCUDAStream();
+ * const auto exec_policy = thrust::cuda::par_nosync(allocator).on(stream);
+ *
+ * Now, one can pass exec_policy to thrust functions
+ *
+ * To get an integer array of size 1000 whose lifetime is managed by unique_ptr,
+ * use:
+ *
+ * auto int_array = allocator.AllocateStorage<int>(1000);
+ *
+ * int_array.get() gives the raw pointer.
+ */
+class CUDAWorkspaceAllocator {
+  using ptr_map_t = std::unordered_map<void*, torch::Tensor>;
+  std::shared_ptr<ptr_map_t> pointers;
+
+ public:
+  using value_type = char;
```

Comment:
Is value_type the keyword to use? otherwise:
```suggestion
  using DataType = char;
```

Reply:
It is required for this class to satisfy allocator requirements for thrust.

Reply:
I see, so this is a keyword required by thrust? If so, add a comment such that people will not remove / rename it accidentally in the future.